### PR TITLE
feat: integrate environment variables for config management

### PR DESCRIPTION
**Pull request by C0nsumption**
- Add .env files in client and server directories
- Implement python-dotenv in Flask server to load environment variables
- Use Vue's dotenv mode to load environment variables in Vue client
- Move sensitive and system-dependent config data to environment variables
- Set Flask's debug mode based on FLASK_ENV environment variable
- Add error handling for environment variables

### CODE REVIEW

File: client/src/components/LoginForm.vue [hunk 1]
```diff
@@ -72,6 +72,9 @@ span {
 import { ref, defineEmits } from 'vue'
 import { useStore } from 'vuex'  // Import useStore
 
+const url = import.meta.env.VITE_API_URL
```
Comment (PropeReferio) [inline client/src/components/LoginForm.vue:75]: Nice, so you didn't hardcode anything dangerous

File: client/src/components/PostSubmission.vue [hunk 2]
```diff
@@ -4,21 +4,56 @@
...
+async function submitPost() {
+  let formData = new FormData()
+  formData.append('title', title.value)
+  formData.append('description', description.value)
+  formData.append('file', file.value)
+
+  const response = await fetch(`${url}/post`, {
+    method: 'POST',
+    headers: {
+      'Authorization': 'Bearer ' + store.state.token
```
Comment (PropeReferio) [inline client/src/components/PostSubmission.vue:47]: Cool, is this stored securely?

File: server/.env.template [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# .env.example
+
+# Your database URI (e.g., sqlite:///site.db)
+DATABASE_URI=<your-database-uri>
+
+# Your secret key for JWT signing and encryption
+SECRET_KEY=<your-secret-key>
```
Comment (PropeReferio) [inline server/.env.template:7]: This is solid documentation, dude.

File: server/main.py [hunk 4]
```diff
@@ -6,9 +6,42 @@
...
 from jwt import exceptions
 
+# Load environment variables from .env file
+load_dotenv()
+
+# Ensure necessary environment variables are set
+required_env_vars = ['DATABASE_URI', 'SECRET_KEY', 'FLASK_ENV']
+missing_env_vars = [var for var in required_env_vars if not os.getenv(var)]
+if missing_env_vars:
+    raise EnvironmentError(f"Required environment variables are missing: {', '.join(missing_env_vars)}")
```
Comment (PropeReferio) [inline server/main.py:20]: Nice

Comment (PropeReferio) [inline server/main.py:20]: Nice

Comment (PropeReferio) [inline server/main.py:20]: Nice

**Review by PropeReferio — APPROVED** [review]

File: client/src/components/PostSubmission.vue (hunk 2 above)
Comment (C0nsumption) [reply client/src/components/PostSubmission.vue:47]: Working on it. Hashing and other auth stuff now. Then unit testing. Thanks for the review :D

**Pull request merged**